packages/flutter_tools/lib/src/base/logger.dart

```diff
@@ -24,67 +33,157 @@ abstract class Logger {
   /// fail in some way.
   void printError(String message, [StackTrace stackTrace]);
 
-  /// Display normal output of the command. This should be used for things like
+  /// Display normal output. This should be used for things like
   /// progress messages, success messages, or just normal command output.
-  void printStatus(String message, { bool emphasis: false, bool newline: true });
+  void printStatus(String message,
+                   { bool emphasis: false, bool newline: true }) {
+    hideStatusLine();
+    printRaw(message, emphasis: emphasis, newline: newline);
+    showStatusLine();
+  }
+
+  /// Display output without saving the status line.  Most people will
+  /// not call this directly.
+  void printRaw(String message, { bool emphasis: false, bool newline: true });
 
   /// Use this for verbose tracing output. Users can turn this output on in order
   /// to help diagnose issues with the toolchain or with their setup.
   void printTrace(String message);
 
-  /// Start an indeterminate progress display.
-  Status startProgress(String message);
-
   /// Flush any buffered output.
   void flush() { }
+
+  Hider commandLine;
+  Status _activeStatus;
+
+  /// Overridden in subclasses to implement progress reports.
+  Status newProgress(String message) {
```

Comment:
It won't let me put the annotation on "newProgress":

"""
Annotation can be only constant variable or constant constructor invocation lib/src/base/logger.dart, line 60
"""